module/behaviour/skills/Getup/src/Getup.cpp

```diff
@@ -61,24 +61,33 @@ namespace module::behaviour::skills {
             // load priorities for the getup
             GETUP_PRIORITY     = config["GETUP_PRIORITY"].as<float>();
             EXECUTION_PRIORITY = config["EXECUTION_PRIORITY"].as<float>();
+            log<NUClear::WARN>("CONFIGURED");
         });
 
-        on<Last<20, Trigger<RawSensors>>, Single>().then(
+        on<Last<250, Trigger<Sensors>>, Single>().then(
```

Comment:
You should only need one since the sensors are already filtered?

Reply:
```cpp
// Average of the last 250 sensors measurments
for (const auto& s : sensors) {
    Hwt = s->Htw.inverse().matrix().cast<double>();
    uZTw += Hwt.rotation().block(0, 2, 3, 1);
    uXTw += Hwt.rotation().block(0, 0, 3, 1);
}
uZTw = uZTw.normalized()
```

Seems that there's an average being taken of the filtered measurements again here?

@TheCoolKidsTable I agree I'm not sure why we're doing so for filtered measurements? Is this to eliminate/dampen the effect of outlier measurements?

Reply:
Yeah the idea is now more if the angle has exceeded FALLEN_ANGLE for the last X msg's, then start the getup.

The main problem I was trying to solve was the getup being executed while it was walking in real life. Will need to investigate in the lab if current odometry solves this already.

Reply:
I also just noticed an issue in sim where if its walking and has delay it doesn't stop trying to walk while on ground.

I will remove the last 250 msg's component and test

Reply:
Yeah 250 messages would cause a delay of almost three seconds before it detects the getup. We get sensors 90 times a second.